### PR TITLE
[15.0][IMP] vault: Add new groups to allow importing/exporting secrets

### PR DESCRIPTION
**Pull request by CarlosRoca13**
cc @Tecnativa TT44749

This RP addresses the issue https://github.com/OCA/server-auth/issues/549

ping @pedrobaeza @fkantelberg @victoralmau

### CODE REVIEW

**Review by pedrobaeza — APPROVED** [review]

File: vault/views/vault_views.xml [hunk 1]
```diff
@@ -43,16 +43,19 @@
...
                         name="action_open_export_wizard"
                         string="Export to file"
+                        groups="vault.group_vault_export"
                     />
                     <button
                         type="object"
                         name="action_open_export_wizard"
                         string="Export to file"
+                        groups="vault.group_vault_export"
                     />
```
Comment (victoralmau) [inline vault/views/vault_views.xml]: I think button is duplicated and should be removed, right?

**Review by victoralmau — APPROVED** [review]

File: vault/views/vault_views.xml (hunk 1 above)
Comment (CarlosRoca13) [reply vault/views/vault_views.xml]: You are right, changed

**Review by victoralmau — APPROVED** [review]

File: vault/models/res_config_settings.py [hunk 2]
```diff
@@ -8,3 +8,9 @@ class ResConfigSettings(models.TransientModel):
     _inherit = "res.config.settings"
 
     module_vault_share = fields.Boolean()
+    group_vault_export = fields.Boolean(
+        "Export Vaults", implied_group="vault.group_vault_export"
+    )
+    group_vault_import = fields.Boolean(
+        "Import Vaults", implied_group="vault.group_vault_import"
+    )
```
Comment (fkantelberg) [inline vault/models/res_config_settings.py:16]: Can you also add it in `vault.res_config_settings_view_form`?

File: vault/security/vault_security.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<odoo>
+    <record id="group_vault_export" model="res.groups">
+        <field name="name">Allow export vaults</field>
```
Comment (fkantelberg) [inline vault/security/vault_security.xml]: `Allow to export vaults`

File: vault/security/vault_security.xml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+        <field name="category_id" ref="base.module_category_hidden" />
+        <field name="users" eval="[(4, ref('base.user_admin'))]" />
+    </record>
+    <record id="group_vault_import" model="res.groups">
+        <field name="name">Allow import vaults</field>
```
Comment (fkantelberg) [inline vault/security/vault_security.xml]: `Allow to import vaults`

**Review by fkantelberg — CHANGES_REQUESTED** [review]

**Comment by CarlosRoca13** [comment]
Changes done!

File: vault/views/res_config_settings_views.xml [hunk 5]
```diff
@@ -19,6 +19,22 @@
                             </div>
                         </div>
                     </div>
+                    <div class="col-xs-12 col-md-6 o_setting_box" id="vault_share">
+                        <div class="o_setting_left_pane">
+                            <field name="group_vault_import" />
+                        </div>
+                        <div class="o_setting_right_pane">
+                            <label for="group_vault_import" />
```
Comment (fkantelberg) [inline vault/views/res_config_settings_views.xml:27]: Could you add a small explanation?

```xml
<div class="text-muted">
    Allow all users to import vaults accessible to them
</div>
```

File: vault/views/res_config_settings_views.xml [hunk 6]
```diff
@@ -19,6 +19,22 @@
...
+                        </div>
+                    </div>
+                    <div class="col-xs-12 col-md-6 o_setting_box" id="vault_share">
+                        <div class="o_setting_left_pane">
+                            <field name="group_vault_export" />
+                        </div>
+                        <div class="o_setting_right_pane">
+                            <label for="group_vault_export" />
```
Comment (fkantelberg) [inline vault/views/res_config_settings_views.xml:38]: Could you add a small explanation?

```xml
<div class="text-muted">
    Allow all users to export vaults accessible to them
</div>
```

**Review by fkantelberg — APPROVED** [review]
Thank you. Would be nice to also add the following explanations too but not mandatory

**Comment by CarlosRoca13** [comment]
Done :)

**Pull request merged**